### PR TITLE
Lock @actions/cache to 3.0.6, later versions break bundler caching

### PR DESCRIPTION
**Pull request by MSP-Greg**
See #483.  I can't look at it until (hopefully) this weekend.

It may be a bug with @actions/cache, as I tried both 3.1.4 & 3.2.1.  Not.sure.

### CODE REVIEW

**Comment by MSP-Greg** [comment]
Two jobs have 429 errors?  That's 'Too Many Requests'?  What else is going to happen today???

See:
https://www.iana.org/assignments/http-status-codes/http-status-codes.xhtml
https://www.rfc-editor.org/rfc/rfc6585.html#section-4

**Comment by MSP-Greg** [comment]
I did some more testing, and it seems that 3.1.0 and 3.1.1 also seem to work.  3.1.2 and later do not.  Also, see:

https://github.com/actions/toolkit/issues/1377

**Comment by simi** [comment]
Thanks for addressing this @MSP-Greg. Would it be enough to restart failing checks for now, :pray: merge and release?

**Comment by eregon** [comment]
My thoughts is we should avoid "blindly" updating to latest versions of dependencies without a need, things tend to break, it's not the first time. Thank you for the PR.

File: yarn.lock [hunk 1]
```diff
@@ -1,428 +1,428 @@
...
-xmlbuilder@~11.0.0:
-  version "11.0.1"
-  resolved "https://registry.yarnpkg.com/xmlbuilder/-/xmlbuilder-11.0.1.tgz#be9bae1c8a046e76b31127726347d0ad7002beb3"
-  integrity sha512-fDlsI/kFEx7gLvbecc0/ohLG50fugQp8ryHzMTuW9vSa1GJ0XYWKnhsUx7oie3G98+r56aTQIUB4kht42R3JvA==
-
-yallist@^4.0.0:
-  version "4.0.0"
-  resolved "https://registry.yarnpkg.com/yallist/-/yallist-4.0.0.tgz#9bb92790d9c0effec63be73519e11a35019a3a72"
-  integrity sha512-3wdGidZyq5PB084XLES5TpOSRA3wjXAlIWMhum2kRcv/41Sn2emQ0dycQW4uZXLejwKvg6EsvbdlVL+FYEct7A==
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (eregon) [inline yarn.lock:1]: This file has CRLF that's why the diff is so big, I'll regenerate it with LF.

**Comment by MSP-Greg** [comment]
> Maybe we can do that by using ruby/setup-ruby twice in in the same workflow job, and having some debug env var or so to fail if the cache isn't restored in the 2nd usage.

Yes, I thought something similar.  I back to testing this because I'm just a little frustrated by it...

**Pull request merged**